### PR TITLE
Delete work branches when done

### LINKED ISSUE #4523: [LakeFSOutputCommitter] Delete work branches when done
This should be configurable for job and for task branches separately.

### PR DESCRIPTION
**Pull request by arielshaqed**
Fixes #4523.

### CODE REVIEW

**Comment by arielshaqed** [comment]
Review & merge this **after #4513**!

**Comment by arielshaqed** [comment]
Ready for review.

Manually tested.

File: clients/hadoopfs/src/main/java/io/lakefs/Constants.java [hunk 1]
```diff
@@ -9,6 +9,8 @@ public class Constants {
     public static final String SECRET_KEY_KEY_SUFFIX = "secret.key";
     public static final String ENDPOINT_KEY_SUFFIX = "endpoint";
     public static final String LIST_AMOUNT_KEY_SUFFIX = "list.amount";
+    public static final String OC_DELETE_JOB_BRANCH = "committer.delete.job_branch";
```
Comment (johnnyaug) [inline clients/hadoopfs/src/main/java/io/lakefs/Constants.java:12]: These configurations can both pass as a branch name configuration.
Would go with something as `committer.cleanup.delete_job_branch` - to clarify it's a boolean.

**Review by johnnyaug — APPROVED** [review]
Looks great! Small comment.

File: clients/hadoopfs/src/main/java/io/lakefs/Constants.java (hunk 1 above)
Comment (arielshaqed) [reply clients/hadoopfs/src/main/java/io/lakefs/Constants.java]: We ended up with these names:

* `spark.hadoop.fs.lakefs.committer.delete_job_branch`
* `spark.hadoop.fs.lakefs.committer.delete_task_branch`

**Review by lynnro314 — APPROVED** [review]
Great, thanks!

**Comment by arielshaqed** [comment]
Thanks!

**Pull request merged**